cli/dcoscli/tail/main.py

```diff
@@ -0,0 +1,369 @@
+"""Output the last part of files in a task's sandbox
+
+Usage:
+    dcos tail --info
+    dcos tail [--follow --completed --lines=N] <task> [<file>]
+
+Options:
+    -h, --help    Show this screen
+    --info        Show a short description of this subcommand
+    --follow      Output data as the file grows
+    --completed   Tail files from completed tasks as well
+    --lines=N     Output the last N lines [default: 10]
+    --version     Show version
+
+Positional Arguments:
+
+    <task>        Only match tasks whose ID matches <task>.  <task> may be
+                  some substring of the ID, or a unix glob pattern.
+
+    <file>        Output this file. [default: stdout]
+"""
+
+import functools
+import sys
+import time
+
+import concurrent.futures
+import dcoscli
+import docopt
+from dcos import cmds, emitting, mesos, util
+from dcos.errors import DCOSException, DefaultError
+
+logger = util.get_logger(__name__)
+emitter = emitting.FlatEmitter()
+
+
+def main():
+    try:
+        return _main()
+    except DCOSException as e:
+        emitter.publish(e)
+        return 1
+
+
+def _main():
+    util.configure_logger_from_environ()
+
+    args = docopt.docopt(
+        __doc__,
+        version="dcos-tail version {}".format(dcoscli.version))
+
+    return cmds.execute(_cmds(), args)
+
+
+def _cmds():
+    """
+    :returns: All of the supported commands
+    :rtype: [Command]
+    """
+
+    return [
+        cmds.Command(
+            hierarchy=['tail', '--info'],
+            arg_keys=[],
+            function=_info),
+
+        cmds.Command(
+            hierarchy=['tail'],
+            arg_keys=['--follow', '--completed', '--lines', '<task>',
+                      '<file>'],
+            function=_tail),
+    ]
+
+
+def _info():
+    """Print tail cli information.
+
+    :returns: process return code
+    :rtype: int
+    """
+
+    emitter.publish(__doc__.split('\n')[0])
+    return 0
+
+
+NO_FILE_EXCEPTION = DCOSException('No files exist.  Exiting.')
+
+
+def _tail(follow, completed, lines, task, path):
+    """ Tail a file in the task's sandbox.
+
+    :param follow: same as unix tail's -f
+    :type follow: bool
+    :param completed: whether to include completed tasks
+    :type completed: bool
+    :param lines: number of lines to print
+    :type lines: int
+    :param task: task pattern to match
+    :type task: str
+    :param path: file path to read
+    :type path: str
+    :returns: process return code
+    :rtype: int
+    """
+
+    if task is None:
+        fltr = ""
+    else:
+        fltr = task
+
+    if path is None:
+        path = 'stdout'
+
+    lines = int(lines)
+
```

Comment:
might as well move this to right above the partial call since you don't need it until then